Library/Homebrew/cmd/bottle.rb

```diff
@@ -300,69 +301,78 @@ def bottle_formula(f)
     puts "./#{filename}"
     puts output
 
-    if ARGV.include? "--rb"
-      File.open("#{filename.prefix}.bottle.rb", "w") do |file|
-        file.write("\# #{f.full_name}\n")
-        file.write(output)
+    if ARGV.include? "--json"
+      json = {
+        f.full_name => {
+          "formula" => {
+            "pkg_version" => f.pkg_version.to_s,
+            "path" => f.path.to_s.gsub("#{HOMEBREW_REPOSITORY}/", ""),
+          },
+          "bottle" => {
+            "disabled?" => f.bottle_disabled?,
+            "disable_reason" => f.bottle_disable_reason,
+            "root_url" => bottle.root_url,
+            "prefix" => bottle.prefix,
+            "cellar" => bottle.cellar.to_s,
+            "revision" => bottle.revision,
+            "tags" => {
+              bottle_tag.to_s => {
+                "filename" => filename.to_s,
+                "sha256" => sha256,
+              },
+            }
+          },
+          "bintray" => {
+            "package" => Bintray.package(f.full_name),
+            "repository" => Bintray.repository(f.tap),
+          },
+        },
+      }
+      File.open("#{filename.prefix}.bottle.json", "w") do |file|
+        file.write Utils::JSON.dump json
       end
     end
   end
 
-  module BottleMerger
-    def bottle(&block)
-      instance_eval(&block)
-    end
-  end
-
   def merge
     write = ARGV.include? "--write"
-    keep_old = ARGV.include? "--keep-old"
 
-    merge_hash = {}
+    bottles_hash = nil
     ARGV.named.each do |argument|
-      bottle_block = IO.read(argument)
-      formula_name = bottle_block.lines.first.sub(/^# /, "").chomp
-      merge_hash[formula_name] ||= []
-      merge_hash[formula_name] << bottle_block
+      bottle_hash = Utils::JSON.load(IO.read(argument))
+      bottles_hash ||= bottle_hash
+      bottles_hash.merge! bottle_hash
```

Comment:
This also applies to a duplicate of this code in `test-bot.rb` in case you haven't noticed yet.